veps/sig-network/decouple-nad.md

```diff
@@ -0,0 +1,178 @@
+
+### Deployment alternatives 
+
+#### Option 1: Deprecate existing injection code and require manual deployment of network-resources-injector
+
+Current KubeVirt functionality will be deprecated and removed within 3 KubeVirt releases.
+Documentation will indicate that network-resource-injector is a required dependency for KubeVirt installations and will refer users to the github repo for installation instructions.
+Relevant CI lanes will be enhanced to perform the deployment. This approach is equivalent to KubeVirt's dependency on Multus and how it is currently deployed. 
+Optionally, log warnings can be added to virt-operator in case network-resources-injector's MutatingWebhookConfiguration is missing, in order to assist in identifying BWC breach associated issues.
+
+Other deployment systems in the KubeVirt group can be considered to assist users in deployment, such as [CNAO](https://github.com/kubevirt/cluster-network-addons-operator), however this proposal's scope is limited to KubeVirt/KubeVirt only.
+
+##### Pros
+- Easiest option to implement and maintain
+- Does not add any code or build steps to current repo
+
+##### Cons
+- Breaks BWC
+
+#### Option 2: Use virt-operator to deploy network-resources-injector
+
+The https://github.com/k8snetworkplumbingwg/network-resources-injector repo includes all the source code, build infrastructure and deployment yamls required for successful deployment of the operator.
+However, it does not have an official published image for the server. 
+1. The server code will be imported into the KubeVirt repository as vendored code [optional - not recommended]
+2. The KubeVirt build system will be enhanced to build the code, and publish the server image.
+  If the external code is not imported, the KubeVirt build system will be enhanced to clone an external repo and build it, in a similar manner to the way CNAO operates.
+3. virt-operator will deploy and reconcile the server along with its service, MutatingWebhookConfiguration, RBAC etc
+ 
+
+##### Pros
+- No BWC breach
+
+##### Cons
+- Virt-operator currently only deploys internal components and does not reconcile 3rd party tools. Letting it deal with external components may architecturally be a slippery slope that may furthermore contribute to blurring responsibilities between KubeVirt, [CNAO](https://github.com/kubevirt/cluster-network-addons-operator) and [HCO](https://github.com/kubevirt/hyperconverged-cluster-operator). 
+It will be hard to explain why it's important to have virt-operator deploy network-resources-injector, but other network dependencies such as multus, sriov-operator/device-plugin, various CNIs etc.
+- Considerably bloat the code base and/or build system, just to replace a few lines of code.
+- "Swallowing" an external project somehow defeats the objective of decoupling dependencies. 
```

Comment:
and prone for doing it again, which is bad practice.
it also miss the point of this effort.